File: Sources/InkFlowKit/Data/DocumentFactory.swift
import Foundation

public enum DocumentFactory {
    public static func create(
        title: String = "",
        body: String = ""
    ) -> InkDocument {
        InkDocument(
            id: UUID(),
            title: title,
            body: body,
            createdAt: .now,
            updatedAt: .now
        )
    }
}

File: src/data/DocumentFactory.ts
import type { InkDocument } from '../models/InkDocument'

export function createDocument(title = '', body = ''): InkDocument {
  const now = Date.now()
  return {
    id: crypto.randomUUID(),
    title,
    body,
    createdAt: now,
    updatedAt: now,
  }
}
